src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -208,7 +209,10 @@ class ProfileClientActor(requestContext: RequestContext) extends Actor with Fire
           RawlsGroupMemberList(userSubjectIds = Some(subjectIds))
         }}
       }
-      case Some(id) => Future.successful(RawlsGroupMemberList(userSubjectIds = Some(Seq(id))))
+      case Some((fcUser, nihUser)) =>
+        if(whitelist contains nihUser) Future.successful(RawlsGroupMemberList(userSubjectIds = Some(Seq(fcUser))))
+        else Future.failed(throw new FireCloudExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden,
```

Comment:
The UI logic is confusing to follow but it looks like these are the conditions:

If your link time has expired, then it will say "Expired"
If you have the 3 valid entries in thurloe but aren't a part of the rawls dbGap group, it will say "Your access will be granted shortly"
If you have the 3 thurloe entries + dbGap group member it will say "Authorized"
For any other condition, it will say "Not Authorized", but I've never actually seen this case happen


You're right that the junk is not inaccurate- because they are technically linked to NIH. It just seems like we'd need a new state for "You're successfully linked to NIH, but you don't have the proper permissions to view TCGA data"